src/Xhgui/Saver.php

```diff
@@ -23,7 +28,20 @@ public static function factory($config)
                 $mongo = new MongoClient($config['db.host'], $config['db.options']);
                 $mongo->{$config['db.db']}->results->findOne();
                 $profiles = new Xhgui_Profiles($mongo->{$config['db.db']});
+		        // Get new profiling ID, if profiler is enabled
+		        if (call_user_func($config['profiler.enable'])) {
+			        self::$lastProfilingId = new MongoId();
```

Comment:
I don't think profiler.enable function should be called for anything except profiler enabling, as we won't know what lies there.

The user should be responsible for starting the profiler or take into account that asking for current profiling id while the profiler isn't running doesn't return anything meaningful (a MongoId that won't be persisted)